enhancements/ingress/set-delete-http-headers.md

```diff
@@ -1002,16 +1006,27 @@ This route can be annotated with `haproxy.router.openshift.io/disable_cookies=tr
   IngressController.
 
 #### Why does a custom header value defined in RouteSpec override IngressControllerSpec value ?
-  Answer: It's quite common to have a global setting with an escape hatch locally to override the global setting.
-  The purpose of this EP is to define custom headers. So, there are two reasons we need such override:
-  1. A route owner shall have the choice to define what headers he wants as per his application requirements(as they might be varying) 
-  than the ones admin has set cluster-wide.
-  2. The IngressController and Route might be owned by the same person who wants to specify some global configuration
-  and then override that global configuration on a small subset of routes. 
+  Answer: In case of HTTP response headers, the headers set in spec.httpHeaders.actions of IngressController will override
+ the headers with same name in Route's spec.httpHeaders.actions or Route annotation.
+
+ In case of HTTP request headers, the headers set in Route using spec.httpHeaders.actions or annotation will override the 
+ headers with the same name present in the IngressController's spec.httpHeaders.actions.
+
+ This is because of the way Haproxy is designed to work. We cannot change the processing order; we're constrained by HAProxy's design: 
+ the frontend receives the request and dispatches it to the backend, and then the frontend gets the 
+ response from the backend.
+
+ In case of HTTP response, Haproxy overrides the header value of the same header name in the backend with
+ the value of the header defined in the frontend. So, [the backend rules are applied first, followed by the frontend's rules.](https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#4.2-http-response)
+ The `http-response set or delete` stanzas are applied and interpreted in this way by the haproxy.
+
+ However, in case of the HTTP request, haproxy overrides the values of the header value of the same header name present in the frontend with the value of the
+ header defined in backend. [The rules are evaluated in their declaration order when they are met in a frontend,
+ listen or backend section.](https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#4.2-http-request)
+ So, the `http-request set or delete` stanzas are applied and interpreted this way by the haproxy.
```

Comment:
Add an example of how to explicitly set the headers for a desired result, both for request and response.  I assume this means that they have to set request headers on Route and corresponding response headers on IC if they want predictability in the entire transaction.

Reply:
they have to set request headers on Route and corresponding response headers ---
You can't set response header in http-request stanzas and vice-versa. haproxy does not allow that.
For example you can set response header X-Frame-Options in request list ad it will get added in http-request stanza which is not allowed by haproxy.

Reply:
Is that what you were testing recently?

Reply:
Yes.....I take my words back.